include/pgduckdb/pgduckdb_utils.hpp

```diff
@@ -19,4 +23,44 @@ TokenizeString(char *str, const char delimiter) {
 	return v;
 };
 
+template <typename T, typename FuncType, typename... FuncArgs>
+std::optional<T>
+PostgresFunctionGuard(FuncType postgres_function, FuncArgs... args) {
```

Comment:
I think we should have a similar function for the other way around, i.e. where we call DuckDB C++ APIs from Postgres C code.